From 09dfb02f9326118651e744ce4939f5d01e69a573 Mon Sep 17 00:00:00 2001
From: nE0sIghT <ykonotopov@gmail.com>
Date: Wed, 5 Aug 2015 13:48:40 +0300
Subject: [PATCH 2/3] No changes, just simplifications to configure code.

Implicitly use SEARCH_LIB in WX_PATH_FIND_LIBRARIES() and WX_FIND_LIB() macros
to avoid having to pass it to them explicitly every time.

Closes #13601.

git-svn-id: https://svn.wxwidgets.org/svn/wx/wxWidgets/trunk@69570 c3d73ce0-8a6f-49c7-b76d-6d57e0e08775

# Conflicts:
#	acinclude.m4
#	configure
#	configure.in
---
 acinclude.m4 | 32 +++++++++++++++++---------------
 configure.in | 16 ++++++++--------
 2 files changed, 25 insertions(+), 23 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index 7f51802..1e307c9 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -43,16 +43,17 @@ for ac_dir in $1 /usr/include;
 ])
 
 dnl ---------------------------------------------------------------------------
-dnl call WX_PATH_FIND_LIBRARIES(search path, lib name), sets ac_find_libraries
-dnl to the full name of the file that was found or leaves it empty if not found
+dnl call WX_PATH_FIND_LIBRARIES(lib name, [optional extra search paths])
+dnl sets ac_find_libraries to the full name of the file that was found
+dnl or leaves it empty if not found
 dnl ---------------------------------------------------------------------------
 AC_DEFUN([WX_PATH_FIND_LIBRARIES],
 [
   ac_find_libraries=
-  for ac_dir in $1;
+  for ac_dir in $2 $SEARCH_LIB
   do
     for ac_extension in a so sl dylib dll.a; do
-      if test -f "$ac_dir/lib$2.$ac_extension"; then
+      if test -f "$ac_dir/lib$1.$ac_extension"; then
         ac_find_libraries=$ac_dir
         break 2
       fi
@@ -116,7 +117,7 @@ AC_DEFUN([WX_LINK_PATH_EXIST],
 ])
 
 dnl ---------------------------------------------------------------------------
-dnl Usage: WX_FIND_LIB(search-path, lib-name, [lib-function to test])
+dnl Usage: WX_FIND_LIB(lib-name, [lib-function to test], [extra search paths])
 dnl
 dnl Tests in a variety of ways for the presence of lib-name
 dnl
@@ -129,23 +130,23 @@ AC_DEFUN([WX_FIND_LIB],
   ac_find_libraries=
 
   dnl Try with pkg-config first. It requires its lib-name parameter lowercase
-  fl_pkgname=`echo "$2" | tr [[:upper:]] [[:lower:]]`
+  fl_pkgname=`echo "$1" | tr [[:upper:]] [[:lower:]]`
   dnl suppress PKG_PROG_PKG_CONFIG output; we don't want to keep seeing it
   PKG_PROG_PKG_CONFIG() AS_MESSAGE_FD> /dev/null
-  PKG_CHECK_MODULES([$2], [$fl_pkgname],
+  PKG_CHECK_MODULES([$1], [$fl_pkgname],
     [
       dnl Start by assuming there are no novel lib paths
       ac_find_libraries="std"
 
-      dnl A simple copy of the internal vars $2_CFLAGS $2_LIBS doesn't work
+      dnl A simple copy of the internal vars $1_CFLAGS $1_LIBS doesn't work
       dnl inside the macro
       dnl
       dnl TODO: When we stop being autoconf 2.61 compatible, the next 2 lines
       dnl should become:
-      dnl AS_VAR_COPY([ac_find_cflags], [$2_CFLAGS])
-      dnl AS_VAR_COPY([fl_libs], [$2_LIBS])
-      eval ac_find_cflags=\$$2_CFLAGS
-      eval fl_libs=\$$2_LIBS
+      dnl AS_VAR_COPY([ac_find_cflags], [$1_CFLAGS])
+      dnl AS_VAR_COPY([fl_libs], [$1_LIBS])
+      eval ac_find_cflags=\$$1_CFLAGS
+      eval fl_libs=\$$1_LIBS
 
       dnl fl_libs may now contain -Lfoopath -lfoo (only non-standard paths are
       dnl added) We only want the path bit, not the lib names
@@ -161,8 +162,8 @@ AC_DEFUN([WX_FIND_LIB],
     [
       if test "x$ac_find_libraries" = "x"; then
         dnl Next with AC_CHECK_LIB, if a test function was provided
-        if test "x$3" != "x"; then
-          AC_CHECK_LIB([$2], [$3], [ac_find_libraries="std"])
+        if test "x$2" != "x"; then
+          AC_CHECK_LIB([$1], [$2], [ac_find_libraries="std"])
         fi
       fi
 
@@ -170,7 +171,8 @@ AC_DEFUN([WX_FIND_LIB],
         dnl Finally try the search path
         dnl Output a message again, as AC_CHECK_LIB will just have said "no"
         AC_MSG_CHECKING([elsewhere])
-        WX_PATH_FIND_LIBRARIES([$1],[$2])
+        dnl $3 will occasionally hold extra path(s) to search
+        WX_PATH_FIND_LIBRARIES([$1], [$3])
         if test "x$ac_find_libraries" != "x"; then
           AC_MSG_RESULT([yes])
         else
diff --git a/configure.in b/configure.in
index 73c3bf1..8252918 100644
--- a/configure.in
+++ b/configure.in
@@ -2767,7 +2767,7 @@ if test "$build" != "$host" -a "$GCC" = yes; then
             x_includes=$ac_find_includes
         fi
         if test -z "$x_libraries" -o "$x_libraries" = NONE; then
-            WX_PATH_FIND_LIBRARIES($SEARCH_LIB, Xt)
+            WX_PATH_FIND_LIBRARIES(Xt)
             x_libraries=$ac_find_libraries
         fi
     fi
@@ -3554,7 +3554,7 @@ libraries returned by 'pkg-config gtk+-2.0 --libs' or 'gtk-config
         dnl test for external libxpm if we're configured to use it
         if test "$wxUSE_GPE" = "yes"; then
             AC_MSG_CHECKING(for gpewidget library)
-            WX_PATH_FIND_LIBRARIES($SEARCH_LIB,gpewidget)
+            WX_PATH_FIND_LIBRARIES(gpewidget)
             if test "$ac_find_libraries" != "" ; then
                 WX_LINK_PATH_EXIST($ac_find_libraries,$GUI_TK_LIBRARY)
                 dnl -lgpewidget must be before all GTK libs and
@@ -3567,7 +3567,7 @@ libraries returned by 'pkg-config gtk+-2.0 --libs' or 'gtk-config
             fi
 
             dnl AC_MSG_CHECKING(for gpe library)
-            dnl WX_PATH_FIND_LIBRARIES($SEARCH_LIB,gpe)
+            dnl WX_PATH_FIND_LIBRARIES(gpe)
             dnl if test "$ac_find_libraries" != "" ; then
             dnl     WX_LINK_PATH_EXIST($ac_find_libraries,$GUI_TK_LIBRARY)
             dnl     GUI_TK_LIBRARY="$GUI_TK_LIBRARY -lgpe"
@@ -3817,7 +3817,7 @@ libraries returned by 'pkg-config gtk+-2.0 --libs' or 'gtk-config
 
 
         AC_MSG_CHECKING(for Motif/Lesstif library)
-        WX_PATH_FIND_LIBRARIES($SEARCH_LIB, Xm)
+        WX_PATH_FIND_LIBRARIES(Xm)
 
         if test "x$ac_find_libraries" != "x" ; then
             AC_MSG_RESULT(found in $ac_find_libraries)
@@ -3956,7 +3956,7 @@ libraries returned by 'pkg-config gtk+-2.0 --libs' or 'gtk-config
         dnl test for external libxpm if we're configured to use it
         if test "$wxUSE_LIBXPM" = "sys"; then
             AC_MSG_CHECKING(for Xpm library)
-            WX_PATH_FIND_LIBRARIES($SEARCH_LIB,Xpm)
+            WX_PATH_FIND_LIBRARIES(Xpm)
             if test "$ac_find_libraries" != "" ; then
                 WX_LINK_PATH_EXIST($ac_find_libraries,$GUI_TK_LIBRARY)
                 GUI_TK_LIBRARY="$GUI_TK_LIBRARY$ac_path_to_link"
@@ -4295,7 +4295,7 @@ if test "$wxUSE_OPENGL" = "yes"; then
             AC_CHECK_HEADER(GL/glu.h, [
                 found_gl=0
 
-                WX_FIND_LIB([$SEARCH_LIB /opt/graphics/OpenGL/lib], GL, glBegin)
+                WX_FIND_LIB(GL, glBegin, [/opt/graphics/OpenGL/lib])
                 if test "$ac_find_libraries" != "" ; then
                     if test "$ac_find_libraries" != "std" ; then
                       WX_LINK_PATH_EXIST([$ac_find_libraries],[$LDFLAGS])
@@ -4307,7 +4307,7 @@ if test "$wxUSE_OPENGL" = "yes"; then
                     dnl don't suppose that libGL and libGLU are always in the
                     dnl same directory -- this is not true for some common
                     dnl distributions
-                    WX_FIND_LIB([$SEARCH_LIB], GLU, gluBeginCurve)
+                    WX_FIND_LIB(GLU, gluBeginCurve, [/opt/graphics/OpenGL/lib])
                     if test "$ac_find_libraries" != "" ; then
                       if test "$ac_find_libraries" != "std" ; then
                         WX_LINK_PATH_EXIST([$ac_find_libraries],[$LDFLAGS])
@@ -4323,7 +4323,7 @@ if test "$wxUSE_OPENGL" = "yes"; then
                 fi
 
                 if test "$found_gl" != 1; then
-                    WX_FIND_LIB([$SEARCH_LIB], MesaGL, glEnable)
+                    WX_FIND_LIB(MesaGL, glEnable, [/opt/graphics/OpenGL/lib])
                     if test "$ac_find_libraries" != "" ; then
                       if test "$ac_find_libraries" != "std" ; then
                         WX_LINK_PATH_EXIST([$ac_find_libraries],[$LDFLAGS])
-- 
2.4.6

